### ISSUE TITLE
Clear-button in filter-modal does not clear filters

### ISSUE THREAD

**Issue by Angamanga** [ushahidi/platform#1953]
### Expected behaviour
When I click on the clear-button, all previously set filters are cleared.
### Actual behaviour
When I click the clear-button, all previously set filters remains.

### Steps to reproduce the behaviour/error
1. Open filter-modal
2. Add a filter
3. Click apply
4. Open filter-modal again
5. Click clear
6. Open filter-modal agan, filters remains

**Comment by Angamanga** [comment]
@Philomenahhfl started on this today, will get handholding from me if needed.

**Comment by Angamanga** [comment]
Tech spec:
- [ ] The cancel-function should clear the filters
- [ ] The cancel-function should close the modal
- [ ] Change the cancel-function to clear the filters by using the post-filter-service

**Comment by jshorland** [comment]
For this particular aspect of filtering, I believe @Angamanga's assessment of expected behavior when clearing filters is correct.

That said, I've been playing around with other tools that use filters (airbnb, intercom, chargebee, etc) and here's my suggestion for how our filtering system SHOULD work (I have put this entirely into new issue # so we can go ahead and push this fix and then address the rest of the issues separately):

1. When I open the filter modal, I want to see ALL things I can filter by, INCLUDING surveys and categories. Currently, surveys and categories are only filterable in the context modal, and though this may seem like a redundancy, I might be creating a complex saved search and I don't want to filter in two places as a user. It's more work for me. put them all in the filter modal, and keep surveys/categories in the context bar as well. 

2. When I click a filter anywhere, I want it to update as I click/unclick. Why? Based on conversations with @justinscherer, we are trying to move to a more efficient way of viewing data in either map or list view instead of clicking back and forth inefficiently. From a similar efficiency perspective, the day may come when a user sees data as they filter it (right now, it's in the background of the filter modal). As a user creating a set of filters to drill down on my data, I want to preview how those filters affect what I'm seeing as I do the action of filtering itself. Then, once I hit "apply", I know for sure that that filter set is what I want to see instead of moving back and forth between a filter modal and data to try to get it just right. (This was inspired by my usage of Airbnb filters and being really annoyed that it didn't update as I clicked a filter on or off, only when i clicked "apply", and then my filters would disappear and I'd have to reopen and hope I got my filtering right the next time). 

3. When I click "clear filters" I just want them all to clear. As a user, i've most likely spent a good amount of time drilling into and browsing my data in a filtered view, and probably clicked on several paths/levels of filters during that journey. "CLEAR" to me means clearing ALL of it. This is where to @justinscherer's point, an "UNDO" action would be helpful. To me, undo means undo'ing my most recent action, and "clear" means undoing all of it.
